atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OraclePrimaryKeyConstraintNames.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.keyvalue.dbkvs.impl.oracle;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.palantir.atlasdb.keyvalue.dbkvs.OracleTableNameMapper;
+
+public final class OraclePrimaryKeyConstraintNames {
+
+    private OraclePrimaryKeyConstraintNames() {
+        //Utility class
+    }
+
+    public static String get(String name) {
+        return truncateToMaxOracleLength("pk_" + name);
+    }
+
+    private static String truncateToMaxOracleLength(String constraintName) {
+        return StringUtils.left(constraintName, OracleTableNameMapper.ORACLE_MAX_TABLE_NAME_LENGTH);
```

Comment:
See above comment about making `getPrimaryKeyConstraintName` protected, but my take would be that this should throw an exception if the generated constraint name exceeds the DB's internal limitations. There have been some nasty issues in the past with silent truncation, so I'd like to avoid those. I realize that making this backwards compatible with previously generated tables & indexes is challenging, so I'd love to see specific tests around exact behavior for existing examples.